presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -154,6 +155,7 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                                         new PushLimitThroughSemiJoin(),
                                         new MergeLimitWithDistinct(),
 
+                                        new PruneSemiJoinColumns(),
```

Comment:
I would group all the new rules which are created out of PruneUnreferencedOutputs in to a single List. 
Then I would add it to this list in the same way as predicate pushdown rules are added.

Thanks to that it will be easy to use these rules in all the places were old rule were used